src/commands/base-command.ts

```diff
@@ -229,38 +203,28 @@ export default class BaseCommand extends Command {
       }
       debug(`${name}:preAction`)('start')
       this.analytics = { startTime: process.hrtime.bigint() }
-      await this.init(actionCommand)
+      await this.init(actionCommand as BaseCommand)
       debug(`${name}:preAction`)('end')
     })
   }
 
-  /** @private */
-  noBaseOptions = false
-
+  #noBaseOptions = false
```

Comment:
Can we not use `private noBaseOptions`?

Reply:
TypeScript private is not a JavaScript private this is truly private and not accessible after transpiling.

TBH I would prefer this notation over using they private keyword as this is just dropped on transpiling

Reply:
Do we care if it's truly private or not, though? I think we should focus on code readability and type safety, and having to prefix every private property with `#` feels a bit awkward and noisy to me, personally.